int64 size
StateData[] data 

